app/src/main/java/de/dhbw/corona_world_app/ui/statistic/StatisticFragment.java

```diff
@@ -74,6 +77,23 @@ public View onCreateView(@NonNull LayoutInflater inflater,
                 ErrorDialog.showBasicErrorDialog(getContext(), ErrorCode.CANNOT_READ_FILE, null);
             }
         }
+        try {
+            Future<Void> future=statisticCallViewModel.getMoreData(StatisticCallDataManager.DataType.ALL_DATA);
+            Future<Void> future1=statisticCallViewModel.getMoreData(StatisticCallDataManager.DataType.FAVOURITE_DATA);
+            future.get();
+            future1.get();
+        } catch (ExecutionException e) {
+            Throwable error=e.getCause();
+            Log.e(TAG,"error getting new Data",e);
+            if(error instanceof IOException){
+                //check if error is undoable
+            }else if(error instanceof DataException){
+                //inform User that data is corrupt and must be remade
+            }
+        }catch (InterruptedException e){
+            Log.e(TAG,"Thread has been interrupted",e);
+            //future.cancel(true);
+        }
```

Comment:
What has the statistic fragment to do with favourite and history?

Reply:
This was hotfix needed in order to initialize the datamanager before submitting a new statisticrequest to the datamanager. If you open a new statistic before initializing the datamanger through history or favourite the app will crash

Reply:
ah, ok. I see